src/lib.rs

```diff
@@ -59,4 +63,57 @@ impl Calc {
             .collect::<Vec<String>>();
         return msgs.join("\n");
     }
+
+    pub fn eval(opcode: AstNode) -> Result<u64, String> {
+        let bytecode = &mut vec![];
+        Self::to_bytecode(opcode, bytecode);
+        return Self::eval_bytecode(bytecode);
+    }
+
+    pub fn to_bytecode(ast_node: AstNode, result: &mut Vec<Instruction>) -> &Vec<Instruction> {
```

Comment:
This function signature is somewhat unexpected.

I'm surprised it builds. It looks like `result` and the return value are the same vector with differing mutability?

Can you get it down to just:

```
pub fn to_bytecode(ast_node: AstNode, prog: &mut Vec<Instruction>) {
```

i.e. no return value? I reckon you can (but I could be wrong).

Reply:
yeah you're right, will rewrite

Reply:
done https://github.com/softdevteam/pavel.yaiwr/pull/10/commits/cfad42a4a0a32de3f6304200c8c350baa19878eb

Reply:
much better!